snap/export_test.go

```diff
@@ -19,7 +19,20 @@
 
 package snap
 
+import (
+	"regexp"
+)
+
 var (
 	ImplicitSlotsForTests        = implicitSlots
 	ImplicitClassicSlotsForTests = implicitClassicSlots
+	oldSupportedHooks            = supportedHooks
 )
+
+func FakeSupportedHookType(pattern *regexp.Regexp) {
```

Comment:
Ah, yes thank you for catching this. I've since noticed that pattern digging into the API, so I will update. Regarding why we need to mock it out at all, I added this mostly because it would be difficult to test for invalid hook names without making the whitelist accept such names in the first place. I also think it makes the tests a little easier to read instead of requiring the reader to know the contents of the whitelist. However, tests within other packages can't use this; they have to use hooks in the whitelist, so it's only a half win in my mind. I'd like to keep the invalid hook name test (unless you think it's unnecessary), but I can probably rewrite these other tests to registered hook types if you prefer.